meet-data/probe.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+# vim: set ts=8 sts=4 et sw=4 tw=99:
+#
+# Library for probes for new meets
+
+from bs4 import BeautifulSoup
+import datetime
+import os
+import shutil
+import sys
+import urllib.request
+
+FEDDIR = os.path.dirname(os.path.realpath(__file__))
+
+def error(msg):
+    print(msg, file=sys.stderr)
+    sys.exit(1)
+
+def gethtml(url):
+    with urllib.request.urlopen(url) as r:
+        return r.read()
+
+def getenteredurls():
+    urls = []
+    for dirname, subdirs, files in os.walk(FEDDIR):
+        if 'URL' in files:
+            with open(dirname + os.sep + 'URL', 'r') as fd:
+                for k in fd.readlines():
+                    urls.append(k.strip())
+    return urls
+
+def main(federation, color, getmeetlist, URL):
```

Comment:
This probably shouldn't be called `main`... It's also doing too much because of the `color`, `getmeetlist`, and `URL` parameters.

It would probably be useful to just have this function take in a `meetlist` and an `enteredmeetlist`, then return a list of meets that haven't been entered. That's a little more complicated than it sounds, because URLs like "foo bar" and "foo%20bar" should compare equal. So they have to be canonicalized, somewhat.

The printing logic should probably be done in the federation's probe script itself, since not all federations have archives that need to be worked through.